packages/@aws-cdk/aws-s3/lib/notifications-resource/notifications-resource-handler.ts

```diff
@@ -128,6 +127,65 @@ const handler = (event: any, context: any) => {
     }
   });
 
+  function getNotificationConfiguration() {
+    const [bucketNotifcations, err] = filteredBucketNotification();
+    if (err) {
+      return err;
+    }
+
+    if (event.RequestType !== 'Delete') {
+      // Add back the updated configuration
+      const inConfig = event.ResourceProperties.NotificationConfiguration;
+      if (inConfig.TopicConfigurations) {
+        bucketNotifcations.TopicConfigurations = inConfig.TopicConfigurations;
+      }
+      if (inConfig.LambdaFunctionConfigurations) {
+        bucketNotifcations.LambdaFunctionConfigurations = inConfig.LambdaFunctionConfigurations;
+      }
+      if (inConfig.QueueConfigurations) {
+        bucketNotifcations.QueueConfigurations = inConfig.QueueConfigurations;
+      }
+    }
+
+    return [bucketNotifcations, null];
+  }
+
+  function filteredBucketNotification() {
+    let error = null;
+    let response: any = {};
+    const params = {
+      Bucket: event.ResourceProperties.BucketName,
+    };
+
+    // Get existing bucket notificaitons and remove the previous or new configuration elements
+    s3.getBucketNotificationConfiguration(params, function(err: any, data: any) {
+      log({ err, data });
+      if (err) {
+        error = err;
+      } else {
+        let inConfig = null;
+        if (event.OldResourceProperties && event.OldResourceProperties.NotificationConfiguration) {
```

Comment:
When fetching the existing S3 bucket notifications, we want to filter out entries that we had configured before for updates or deleted.  Or filter out new entries when adding a new set of notifications